org.osate.ui/src/org/osate/ui/wizards/NewAadlPackageWizard.xtend

```diff
@@ -68,6 +81,50 @@ class NewAadlPackageWizard extends AbstractNewFileWizard {
 		]
 	}
 	
+	def private findPackageInScope(IContainer parent, String packageName) {
+		/* Parent might be a Project, which causes problems below, so let's append
+		 * a bogus folder to it.
+		 */
+		val IFolder fakeFolder = parent.getFolder(Path.forPosix(".fake"))
+		val Resource rsrc = OsateResourceUtil.getResource(fakeFolder)
+		val scope = globalScopeProvider.getScope(rsrc, Aadl2Package.eINSTANCE.getPackageRename_RenamedPackage(), null)
+		val qualifiedName = qNameConverter.toQualifiedName(packageName);
+		return scope.getSingleElement(qualifiedName)
+	}
+		
+	// Not going to get here if no project is selected
+	override String validateFileName(IContainer parent, String packageName) {
+		val found1 = findPackageInScope(parent, packageName)
+		if (found1 !== null) {
+			val foundFile = OsateResourceUtil.getOsateIFile(found1.EObjectURI)
+			val foundProject = foundFile.getProject()
+			if (foundProject === parent.getProject()) {
+				return "Package '" + packageName + "' already exists in the selected project: '" + foundFile.projectRelativePath + "'"
+			} else {
+				return "Package '" + packageName + "' already exists in project '" + foundProject.name + "' that the selected project depends on: '" + foundFile.projectRelativePath + "'"
+			}
+		} else {
+			/* See if the package exists in scope in any project that depends on the current project.
+			 * This isn't an error for the current project, but it will mess up other projects.
+			 */
+			for (user : parent.getProject().referencingProjects) {
+				val found2 = findPackageInScope(user, packageName)
+				if (found2 !== null) {
+					val foundFile = OsateResourceUtil.getOsateIFile(found2.EObjectURI)
+					val foundProject = foundFile.getProject()
+					if (foundProject != user) {
+						return "Package '" + packageName + "' already exists in project '" + foundProject.name + "' that is depended on by project '" + user.name + "' that depends on the selected project: '" + foundFile.projectRelativePath + "'"
```

Comment:
The most round-about case.  The named package exists in project `C` that project `B` depends on, and proejct `B` also depends on the currently selected project.  Not a problem for the current project, but breaks project `B`.